SYCL/AtomicRef/add.cpp

```diff
@@ -2,10 +2,11 @@
 // RUN: -o %t.out %gpu_atomics_config
 // RUN: %HOST_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
-// TODO: Drop the separate "emulation" compilation once "native" OpenCL CPU
-// support is provided.
+// TODO: Drop the separate "non-native" runs once native OpenCL CPU
+// and CUDA support is provided.
```

Comment:
I would:
- avoid using %gpu_atomics_config variable - it's value hard to predict by reading test source
- use explicit compiler option instead
- split add.cpp and sub.cpp into two sets of tests add-native.cpp and add-emulated.cpp

Reply:
> Avoid using %gpu_atomics_config variable - it's value hard to predict by reading test source, use explicit compiler option instead
In the current layout, this is the only way to handle Intel GPUs vs Nvidia automatically. 
However, if I was to take the next advice (and I will):
> split add.cpp and sub.cpp into two sets of tests add-native.cpp and add-emulated.cpp
...then I could just mark the native test as `UNSUPPORTED` for CUDA.

Reply:
Done. I guess I should also switch to using common source files for both test types.

Reply:
It might be useful if you expect sources be changed before all devices natively support this feature. Otherwise, it looks okay to me.